### PR TITLE
refactor sources to source pkg

### PR DESCRIPTION
**Pull request by andyxning**
This PR refactor different sources to `sources` package in case we will add more sources in the future.

/cc @davidopp @dchen1107 @balajismaniam @ConnorDoyle

### CODE REVIEW

**Comment by ConnorDoyle** [comment]
I can get to this post v1.8
In general, I think this is a good refactor

**Comment by andyxning** [comment]
@ConnorDoyle Thanks for the reply. Hope you guys take a look at this once you have time.

File: source/cpuid/cpuid.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cpuid
+
+import "github.com/klauspost/cpuid"
+
+// CPUIDSource implements FeatureSource.
+type CPUIDSource struct{}
```
Comment (ConnorDoyle) [inline source/cpuid/cpuid.go]: Could we rename this to just `Source`? That way the qualified name is `cpuid.Source` which sounds a bit better to me.

(Applies to the other feature sources too.)

Comment (andyxning) [reply source/cpuid/cpuid.go]: @ConnorDoyle Will do.

**Comment by andyxning** [comment]
@ConnorDoyle Done. PTAL.

**Comment by ConnorDoyle** [comment]
Merge bot has gone insane, manually merged.

**Comment by balajismaniam** [comment]
@ConnorDoyle  @andyxning Sorry about the merge bot spams. It should stop now. I've disabled it.

**Pull request merged**